internal/server/v2/observation/calculation_util_test.go

```diff
@@ -152,3 +154,194 @@ func TestFindObservationResponseHoles(t *testing.T) {
+								Date:  "2",
+								Value: proto.Float64(2),
+							}},
+						},
+					}},
+				}},
+			},
+			Facets: map[string]*pb.Facet{
+				"1": {
+					ObservationPeriod: "P1M",
+				},
+				"2": {
+					MeasurementMethod: "US_Census",
+					ObservationPeriod: "P1Y",
+				},
+			},
+		},
+		&VariableFormula{
+			LeafData: map[string]*ASTNode{
+				"Count_Person": {StatVar: "Count_Person"},
+				"Count_Person[mm=US_Census;p=P1Y]": {
+					StatVar: "Count_Person",
+					Facet: &pb.Facet{
+						MeasurementMethod: "US_Census",
+						ObservationPeriod: "P1Y",
+					},
+				},
+			},
+		},
+		&VariableFormula{
+			LeafData: map[string]*ASTNode{
+				"Count_Person": {
+					StatVar: "Count_Person",
+					CandidateSeries: map[string]map[string][]*pb.PointStat{
+						"geoId/01": {
+							"1": {{
+								Date:  "1",
+								Value: proto.Float64(1),
+							}},
+							"2": {{
+								Date:  "2",
+								Value: proto.Float64(2),
+							}},
+						},
+					},
+				},
+				"Count_Person[mm=US_Census;p=P1Y]": {
+					StatVar: "Count_Person",
+					Facet: &pb.Facet{
+						MeasurementMethod: "US_Census",
+						ObservationPeriod: "P1Y",
+					},
+					CandidateSeries: map[string]map[string][]*pb.PointStat{
+						"geoId/01": {
+							"2": {{
+								Date:  "2",
+								Value: proto.Float64(2),
+							}},
+						},
+					},
+				},
+			},
+		},
+	}} {
+		computeLeafSeries(c.inputResp, c.formula)
+		if ok := reflect.DeepEqual(c.formula, c.want); !ok {
+			t.Errorf("computeLeafSeries = %v, want %v",
+				c.formula, c.want)
+		}
+	}
+}
+
+func TestEvalBinaryExpr(t *testing.T) {
+	for _, c := range []struct {
+		x    map[string]map[string][]*pb.PointStat
+		y    map[string]map[string][]*pb.PointStat
+		op   token.Token
+		want map[string]map[string][]*pb.PointStat
+	}{
+		{
+			map[string]map[string][]*pb.PointStat{
+				"geoId/01": {"facetId2": {
+					{
+						Date:  "1",
+						Value: proto.Float64(1),
+					},
+					{
+						Date:  "3",
+						Value: proto.Float64(3),
+					},
+					{
+						Date:  "4",
+						Value: proto.Float64(4),
+					},
+					{
+						Date:  "5",
+						Value: proto.Float64(5),
+					},
+					{
+						Date:  "8",
+						Value: proto.Float64(8),
+					},
+				}},
+				"geoId/02": {"facetId1": {{
+					Date:  "1",
+					Value: proto.Float64(1),
+				}}},
+			},
+			map[string]map[string][]*pb.PointStat{
+				"geoId/01": {
+					"facetId1": {{
+						Date:  "1",
+						Value: proto.Float64(1),
+					}},
+					"facetId2": {
+						{
+							Date:  "2",
+							Value: proto.Float64(2),
+						},
+						{
+							Date:  "3",
+							Value: proto.Float64(3),
+						},
+						{
+							Date:  "5",
+							Value: proto.Float64(5),
+						},
+						{
+							Date:  "6",
+							Value: proto.Float64(6),
+						},
+						{
+							Date:  "7",
+							Value: proto.Float64(7),
+						},
+					}},
+				"geoId/02": {"facetId1": {{
+					Date:  "1",
+					Value: proto.Float64(1),
+				}}},
+			},
+			token.ADD,
```

Comment:
do you want to have test for other token?

Reply:
Added tests for all 4 tokens